### PR TITLE
Add `/aws/assume/$role` endpoint

### PR DESCRIPTION
**Pull request by maruina**
This PR introduces a breaking change. The AWS endpoint now takes two subpaths:
- `/aws/metadata/$path` has the old `/aws/$path` feature to access AWS metadata
- `/aws/assume/$role_arn` tells go-infrabin to try to assume the AWS role calling STS Assume Role

In addition:
- remove `dependabot` as superseeded by Renovate
- remove the duplicated `tools.go` file
- fix linting errors
- configure coveralls to not report a variation smaller than 5%

### CODE REVIEW

File: internal/aws/clients.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+			AssumedRoleId: aws.String(fakeAssumedRoleId),
+		},
+		Credentials: &types.Credentials{
+			SecretAccessKey: aws.String(fakeSecretAccessKey),
+			SessionToken:    aws.String(fakeSessionToken),
+			Expiration:      aws.Time(timeNow()),
+			AccessKeyId:     aws.String(fakeAccessKeyId),
+		},
+	}, nil
+}
```
Comment (janeklb) [inline internal/aws/clients.go:49]: Can this be moved into a test package / module / file?

Comment (janeklb) [reply internal/aws/clients.go:49]: no i mean into something totally separate -- just for tests.. maybe in test helper package/ module?

**Review by Smirl — CHANGES_REQUESTED** [review]
Should update the README?

File: README.md [hunk 2]
```diff
@@ -156,6 +156,17 @@ See the [README](./chart/go-infrabin/README.md).
 
   * _returns_: JSON of the requested path
 
+* _grpc_: `infrabin.Infrabin.AWSAssume` _rest_: `GET /aws/assume/<path>`
```
Comment (Smirl) [inline README.md:159]:
```suggestion
* _grpc_: `infrabin.Infrabin.AWSAssume` _rest_: `GET /aws/assume/<role>`
```

File: README.md [hunk 3]
```diff
@@ -156,6 +156,17 @@ See the [README](./chart/go-infrabin/README.md).
...
+  * _grpc request_
+
+  ```text
+  message AWSAssume {
+      string role = 1;
+  }
+  ```
+
+  * _returns_: JSON with the AssumedRoleId from AWS
```
Comment (Smirl) [inline README.md:168]: We give examples for other known response types. a little example would be nice

**Review by Smirl — APPROVED** [review]
LGTM

**Pull request merged**